### PR TITLE
Add Github Action

### PR DESCRIPTION
**Pull request by patjouk**
This PR adds GHA.
The GHA config file is a 1:1 conversion of the existing [`azure-pipelines.yml`](https://github.com/explosion/catalogue/blob/master/azure-pipelines.yml). You can see the [action running here](https://github.com/patjouk/catalogue/actions/runs/4133234175).

Let me know if you want any change :)

CC @ryndaniels for visibility

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  tests:
+    name: Test
+    strategy:
+      fail-fast: true
```
Comment (honnibal) [inline .github/workflows/tests.yml:12]: Do we want `fail-fast: true`? I think on the other discussion I saw we wanted `fail-fast: false`.

The reason we usually do that is that it's nice to know whether the failure is build-specific, or if it's the code. In most simple bugs all the tests will be red and that's informative.

Comment (adrianeboyd) [reply .github/workflows/tests.yml:12]: My suggestion has been to use `false` for all the smaller repos with fast build+tests. `true` might make sense for `thinc` or `spacy`, though?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        run: |
+          python -m pip install -U build pip setuptools
+          python -m pip install -U -r requirements.txt
+          python -m build --sdist
+
+      - name: Run mypy
+        run: python -m mypy catalogue
+
+      - name: Delete source directory
+        uses: JesseTG/rm@v1.0.3
```
Comment (adrianeboyd) [inline .github/workflows/tests.yml:39]: It feels weird to depend on some random person's action here. Is there no built-in option?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
+
+      - name: Configure Python version
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python_version }}
+          architecture: x64
+          cache: "pip"
```
Comment (adrianeboyd) [inline .github/workflows/tests.yml:27]: We're also still not sure of the side effects here.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (patjouk) [reply .github/workflows/tests.yml:12]: That's on me: I used it for debug and forgot to switch it back. Will push a change soon.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (patjouk) [reply .github/workflows/tests.yml:27]: I'll remove the cache

File: .github/workflows/tests.yml (hunk 2 above)
Comment (patjouk) [reply .github/workflows/tests.yml:39]: I followed what Ines was doing here for standardization reason: https://github.com/explosion/radicli/blob/main/.github/workflows/test.yml#L26-L29
I can look into something else if you prefer.

**Comment by adrianeboyd** [comment]
Looks like everything has been addressed, thanks!

**Pull request merged**